draft-js-plugins-editor/CHANGELOG.md

```diff
@@ -14,6 +14,12 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 
 - `decorators` don't decorate plugins anymore.
 
+### Changed
+
+- *Breaking change* The `plugins` and `decorators` editor props need to be an Immutable.List()
```

Comment:
I compare both the plugins and decorators, not sure if that is what you meant.

The proposed code looks nice as well, only downside I see that now you have two different places to work with plugins. Or do you want this to replace the plugins and decorators properties so you always set them outside the editor? 

I would like to "hide" this knowledge from the user, you set plugins via props then I expect them to update when the props value updates, this is how React generally works.